File: src/main/java/com/matera/restserver/exception/EntityNotFoundException.java
package com.matera.restserver.exception;

import com.matera.restserver.util.Messages;

public class EntityNotFoundException extends Exception {

	private static final String DEFAULT_MESSAGE = "exceptions.default.entitynotfoundexception";

	private static final long serialVersionUID = 8032193805676955555L;

	public EntityNotFoundException(String msg) {
		super(msg);
	}

	public EntityNotFoundException() {
		super(Messages.getMessage(DEFAULT_MESSAGE));
	}

}

File: src/main/kotlin/com/matera/restserver/exception/EntityNotFoundException.kt
package com.matera.restserver.exception

import com.matera.restserver.util.Messages.getMessage
import lombok.Generated

@Generated
class EntityNotFoundException : Exception {
    constructor(msg: String?) : super(msg)
    constructor() : super(getMessage(DEFAULT_MESSAGE))

    companion object {
        private const val DEFAULT_MESSAGE = "exceptions.default.entitynotfoundexception"
        private const val serialVersionUID = 8032193805676955555L
    }
}
